### PR TITLE
Aravis bump version

### PR DESCRIPTION
**Pull request by weatherhead99**
Specify library name and version:  **aravis/0.8.29**

This bumps the dependency versions and the supplied version of aravis. The latest upstream version is 0.8.29. There is also an issue request for an earlier version (here  #20338 ) but that is now an old version and per FAQ I have not added it. Maybe I can if @fixstars-sonali has a specific need/reason? 

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by fixstars-sonali** [comment]
@weatherhead99 I need aravis/0.8.25 as it is the latest version that supports GenDC data format (supported by U3V 1.1)

**Comment by weatherhead99** [comment]
@fixstars-sonali that's very reasonable! On the other hand, as per https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/sources_and_patches.md#removing-old-versions though if they don't bump the major version it'll eventually get tagged for removal I guess. Maybe we can add both versions (I am using aravis at latest version so want it too), and mark 0.8.25 to keep because of the GenDC issue

**Comment by weatherhead99** [comment]
@fixstars-sonali hopefully you don't mind, I've cherry-picked from your PR into this one.

**Comment by fixstars-sonali** [comment]
@weatherhead99 Thank you. 

> @fixstars-sonali hopefully you don't mind, I've cherry-picked from your PR into this one.

Yes, You can use the changes. Thank you.

**Comment by RubenRBS** [comment]
Leaving a comment in the config.yml to specify why a version should stay is always a good idea and we mostly lsiten to it! :)
I've done so now along with some extra change needed for zlib

Thank you both @weatherhead99 and @fixstars-sonali for your contributions :)
@fixstars-sonali, could you please sign the CLA so that this PR can be merged? Thanks!

**Comment by RubenRBS** [comment]
Seems like v2 has a problem https://c3i.jfrog.io/c3i/misc-v2/summary.html?json=https://c3i.jfrog.io/c3i/misc-v2/logs/pr/20354/8-windows-msvc/aravis/0.8.20//summary.json

**Comment by weatherhead99** [comment]
I spun up a windows machine to mess around with this, I can successfully get it to build but ONLY if we have meson/1.2.1 or lower,  meson/1.2.2 fails. Opened ticket in mesonbuild. In the meantime I can't see why this is now failing on the CI??

**Comment by weatherhead99** [comment]
@RubenRBS  all the v1 builds are now green, it won't let me see the v2 build issues....

**Comment by weatherhead99** [comment]
actually is this to do with the fact that I should have a check for the conf item that encodes pkg-config binary location? 
Not sure how I do that when I'm actually only indirectly depending on it anyway.

**Comment by AndreyMlashkin** [comment]
> actually is this to do with the fact that I should have a check for the conf item that encodes pkg-config binary location? Not sure how I do that when I'm actually only indirectly depending on it anyway.

you can simply re-trigger the build by closing and reopening the PR or just click "update branch", it will merge master into it and trigger the build

**Comment by AndreyMlashkin** [comment]
@RubenRBS it's quite frustrating for the contributers to get such errors. The process is slow by itself and it makes it last months sometimes

**Comment by RubenRBS** [comment]
> @RubenRBS it's quite frustrating for the contributers to get such errors. The process is slow by itself and it makes it last months sometimes

Thanks for the feedback, we're aware of this and are not sitting idly on it, but more pressing things have been taking our time lately so we have had little time to focus on these kinds of things - We'll bump the priority of this one, see if something can be done :)

File: recipes/aravis/all/conanfile.py [hunk 1]
```diff
@@ -87,10 +103,14 @@ def validate(self):
...
+        if self.settings.os == "Windows":
+            self.tool_requires("meson/[<1.2.2]")
+        else:
+            self.tool_requires("meson/1.2.2")
         self.tool_requires("glib/<host_version>")
         if not self.conf.get("tools.gnu:pkg_config", check_type=str):
-            self.tool_requires("pkgconf/1.9.5")
+            self.tool_requires("pkgconf/2.0.3")
         if self.options.introspection:
             self.tool_requires("gobject-introspection/1.72.0")
```
Comment (RubenRBS) [inline recipes/aravis/all/conanfile.py:115]: The cherry-picked PR has this moved to a normal requires. I'm thinking it was an error on the original PR, but I'd like to double check first :)

**Review by RubenRBS — COMMENTED** [review]
Thanks! I've now relaunched the build

**Comment by RubenRBS** [comment]
@fixstars-sonali also, if you depend on the specific version in production, please do read https://docs.conan.io/2/devops/using_conancenter.html which goes into some detail as to how we recommend using CCI in such environments :)

File: recipes/aravis/all/conandata.yml [hunk 2]
```diff
@@ -1,7 +1,23 @@
 sources:
-  "0.8.20":
-    url: "https://github.com/AravisProject/aravis/releases/download/0.8.20/aravis-0.8.20.tar.xz"
-    sha256: "0c0eb5a76109f29180c09c7e6a23fd403633bf22bbe8468a0ae44995c4449f46"
+    "0.8.29":
+        url: "https://github.com/AravisProject/aravis/releases/download/0.8.29/aravis-0.8.29.tar.xz"
+        sha256: "12e5f2f0e1a966c3a6dce0a42d96b2f24497f42ae6051d6f026811124e986963"
+    "0.8.25":
```
Comment (RubenRBS) [inline recipes/aravis/all/conandata.yml:5]:
```suggestion
    "0.8.25":  # latest version that supports GenDC data format (supported by U3V 1.1)
```

File: recipes/aravis/all/conanfile.py [hunk 3]
```diff
@@ -52,6 +54,9 @@ def config_options(self):
             del self.options.fPIC
         if self.settings.os not in ["Linux", "FreeBSD"]:
             del self.options.packet_socket
+        if Version(self.version).patch < 25:
```
Comment (RubenRBS) [inline recipes/aravis/all/conanfile.py:57]:
```suggestion
        if Version(self.version) < "8.25":
```

File: recipes/aravis/all/conanfile.py [hunk 4]
```diff
@@ -87,10 +103,14 @@ def validate(self):
             )
 
     def build_requirements(self):
-        self.tool_requires("meson/1.2.1")
+        #windows build: meson/1.2.1 works, meson/1.2.2 breaks for some reason!
+        if self.settings.os == "Windows":
+            self.tool_requires("meson/[<1.2.2]")
+        else:
+            self.tool_requires("meson/1.2.2")
```
Comment (RubenRBS) [inline recipes/aravis/all/conanfile.py:110]:
```suggestion
        self.tool_requires("meson/1.2.1")
```

Let's keep it pinned for now - We expect to enable tool require version ranges soon, but this is a better fit for now

File: recipes/aravis/all/patches/0.8.25-gst-shared-lib.patch [hunk 5]
```diff
@@ -0,0 +1,11 @@
+--- a/gst/meson.build	2023-10-03 14:55:30.481822400 -0700
++++ b/gst/meson.build	2023-10-03 14:56:40.939480100 -0700
+@@ -17,7 +17,7 @@
+ 
+ gst_plugin_filename = 'gstaravis.@0@'.format (aravis_api_version)
+ 
+-gst_plugin = shared_library (gst_plugin_filename,
++gst_plugin = library (gst_plugin_filename,
```
Comment (RubenRBS) [inline recipes/aravis/all/patches/0.8.25-gst-shared-lib.patch:8]: Can I get some context into this patch @fixstars-sonali? Thanks!

File: recipes/aravis/all/conanfile.py (hunk 1 above)
Comment (weatherhead99) [reply recipes/aravis/all/conanfile.py:112]: yes, on windows we need pkg-config to find glib.

File: recipes/aravis/all/patches/0.8.25-gst-shared-lib.patch (hunk 5 above)
Comment (weatherhead99) [reply recipes/aravis/all/patches/0.8.25-gst-shared-lib.patch:8]: this allows us to find the gst plugins whether they're built as shared or static. Most people on a "normal" system would have them shared but of course in conan everything is static by default, that's my guess

**Review by AndreyMlashkin — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**